### PR TITLE
JAVA-10379 Fix tests in hibernate-mapping-2 module

### PR DESCRIPTION
**Pull request by anuragkumawat**

### CODE REVIEW

File: persistence-modules/hibernate-mapping-2/src/test/java/com/baeldung/hibernate/manytomany/HibernateManyToManyAnnotationMainIntegrationTest.java [hunk 1]
```diff
@@ -23,60 +23,66 @@
...
         @SuppressWarnings("unchecked")
-        List<Employee> employeeList = session.createQuery("FROM Employee").list();
+		List<Project> projectList = session.createQuery("FROM Project").list();
         assertNotNull(employeeList);
+        assertNotNull(projectList);
+        assertEquals(2, employeeList.size());
+        assertEquals(2, projectList.size());
+        
         for(Employee employee : employeeList) {
             assertNotNull(employee.getProjects());
```
Comment (kwoyke) [inline persistence-modules/hibernate-mapping-2/src/test/java/com/baeldung/hibernate/manytomany/HibernateManyToManyAnnotationMainIntegrationTest.java:54]: Assert here that every employee has 2 projects assigned.

File: persistence-modules/hibernate-mapping-2/src/test/java/com/baeldung/hibernate/manytomany/HibernateManyToManyAnnotationMainIntegrationTest.java [hunk 2]
```diff
@@ -23,60 +23,66 @@
...
 
-    @AfterClass
-    public static void afterTests() {
-        sessionFactory.close();
-    }
+		for (String proj : projectData) {
+			projects.add(new Project(proj));
+		}
+
+		for (String emp : employeeData) {
```
Comment (kwoyke) [inline persistence-modules/hibernate-mapping-2/src/test/java/com/baeldung/hibernate/manytomany/HibernateManyToManyAnnotationMainIntegrationTest.java:70]: Here you only assign  projects to employees, but I think you should also take care of the opposite direction manually - assign employees to projects.

File: persistence-modules/hibernate-mapping-2/src/test/java/com/baeldung/hibernate/manytomany/HibernateManyToManyAnnotationMainIntegrationTest.java (hunk 1 above)
Comment (kwoyke) [inline persistence-modules/hibernate-mapping-2/src/test/java/com/baeldung/hibernate/manytomany/HibernateManyToManyAnnotationMainIntegrationTest.java:54]: Assert here that every employee has 2 projects assigned.

File: persistence-modules/hibernate-mapping-2/src/test/java/com/baeldung/hibernate/manytomany/HibernateManyToManyAnnotationMainIntegrationTest.java [hunk 3]
```diff
@@ -23,60 +23,66 @@
...
         assertNotNull(employeeList);
+        assertNotNull(projectList);
+        assertEquals(2, employeeList.size());
+        assertEquals(2, projectList.size());
+        
         for(Employee employee : employeeList) {
             assertNotNull(employee.getProjects());
         }
+        for(Project project : projectList) {
+            assertNotNull(project.getEmployees());
```
Comment (kwoyke) [inline persistence-modules/hibernate-mapping-2/src/test/java/com/baeldung/hibernate/manytomany/HibernateManyToManyAnnotationMainIntegrationTest.java:57]: Here you should assert that every project has assigned employees.

Comment (anuragkumawat) [reply persistence-modules/hibernate-mapping-2/src/test/java/com/baeldung/hibernate/manytomany/HibernateManyToManyAnnotationMainIntegrationTest.java:57]: project. getEmployees() is returning an empty list. There might be some issue in manytomany mapping.

File: persistence-modules/hibernate-mapping-2/src/test/java/com/baeldung/hibernate/manytomany/HibernateManyToManyAnnotationMainIntegrationTest.java (hunk 1 above)
Comment (anuragkumawat) [reply persistence-modules/hibernate-mapping-2/src/test/java/com/baeldung/hibernate/manytomany/HibernateManyToManyAnnotationMainIntegrationTest.java:54]: This has been addressed

File: persistence-modules/hibernate-mapping-2/src/test/java/com/baeldung/hibernate/manytomany/HibernateManyToManyAnnotationMainIntegrationTest.java (hunk 3 above)
Comment (anuragkumawat) [reply persistence-modules/hibernate-mapping-2/src/test/java/com/baeldung/hibernate/manytomany/HibernateManyToManyAnnotationMainIntegrationTest.java:58]: This has been addressed

File: persistence-modules/hibernate-mapping-2/src/test/java/com/baeldung/hibernate/manytomany/HibernateManyToManyAnnotationMainIntegrationTest.java (hunk 2 above)
Comment (anuragkumawat) [reply persistence-modules/hibernate-mapping-2/src/test/java/com/baeldung/hibernate/manytomany/HibernateManyToManyAnnotationMainIntegrationTest.java:72]: This has been addressed

**Pull request merged**